pkgs/test/test/runner/signal_test.dart

```diff
@@ -181,41 +181,6 @@ void main() {
       await signalAndQuit(test);
     });
 
-    test('causes expect() to always throw an error immediately', () async {
```

Comment:
@jakemac53 - I had to delete this test. This is a user visible behavior difference.

I don't think this is very important. This only impacts behavior after a user sends a `^C`, and then only causes new exceptions (which might interfere with cleanup).

Let me know if you have concerns.

Reply:
This was originally handled by

https://github.com/dart-lang/test/blob/61a8888294937bb7a909f161faaca3052102050c/pkgs/test_api/lib/src/frontend/expect.dart#L93

I don't think that it's a good idea to leak the idea of "closed" into the new API. As I discovered in https://github.com/dart-lang/test/pull/1481 the concept of "closed" is a little fuzzy and brings in details about how the zones are structured in the test runner.

The changed behavior is pretty minor, and should only be hit in unusual cases.

Reply:
I don't actually understand the test.... it looks like it was expecting to hit the `catch` but I don't see how....?

Exiting the tests early is somewhat nice potentially but I agree it isn't critical or anything.

Reply:
> it looks like it was expecting to hit the `catch`

Yes it was.

> but I don't see how....?

It hit the catch because `expect` used to have this line:

```dart
 if (Invoker.current!.closed) throw ClosedException(); 
 ```
 
 The invoker gets marked as closed on the first signal received. I didn't want to expose a `closed` boolean in `hooks.dart`, so expect implementations can't do this anymore.

> I agree it isn't critical or anything.

I suspect no one will even notice. If it does cause problems we can definitely revisit the decision.

Reply:
Ah ok ya that behavior seems super weird. The change LGTM